app/Http/Controllers/Group/WorkspaceController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Http\Controllers\Group;
+
+use App\Http\Controllers\Controller;
+use App\Models\Bookmark;
+use App\Models\Group;
+use App\Models\LegiScan\Session;
+use App\Models\Workspace;
+
+class WorkspaceController extends Controller
+{
+    public function show(Group $group, Workspace $workspace)
+    {
+        $sessionYears = Session::validYears();
+
+        $bookmarks = Bookmark::query()
+                        ->perWorkspace($workspace)
```

Comment:
As long as you're going to all the trouble to get session years out of here: I would make yourself a ticket to create 3 methods:

```
group->findBookmarks();
workspace->findBookmarks();
topic->findBookmarks();
```

since the only reason you're doing it all inline now is because you need to know `per<Model>(<model>)`. If you define it on all 3 models, that problem is solved.

Reply:
Done